internal/cmd/logout_test.go

```diff
@@ -43,16 +49,70 @@ func TestLogout(t *testing.T) {
 			Version: "0.3",
 			Hosts: []ClientHostConfig{
 				{
-					Name:          "host1",
+					Name:          "user1",
 					Host:          srv.Listener.Addr().String(),
 					AccessKey:     "the-access-key",
+					PolymorphicID: "pid1",
 					SkipTLSVerify: true,
+					Current:       true,
 				},
 				{
-					Name:          "host2",
+					Name:          "user2",
 					Host:          srv2.Listener.Addr().String(),
 					AccessKey:     "the-access-key",
+					PolymorphicID: "pid2",
+					SkipTLSVerify: true,
+				},
+			},
+		}
+		err := writeConfig(&cfg)
+		assert.NilError(t, err)
+
+		kubeCfg := clientcmdapi.Config{
+			Clusters: map[string]*clientcmdapi.Cluster{
+				"keep:not-infra": {Server: "https://keep:8080"},
+				"infra:prod":     {Server: "https://infraprod:8080"},
+			},
+			Contexts: map[string]*clientcmdapi.Context{
+				"keep:not-infra": {Cluster: "keep:not-infra"},
+				"infra:prod":     {Cluster: "infra:prod"},
+			},
+			AuthInfos: map[string]*clientcmdapi.AuthInfo{
+				"keep:not-infra": {Token: "keep-token"},
+				"infra:prod":     {Token: "infra-token"},
+			},
+		}
+		err = clientcmd.WriteToFile(kubeCfg, kubeConfigPath)
+		assert.NilError(t, err)
+		return cfg, &count
+	}
+
+	setupUnauthorized := func(t *testing.T) (ClientConfig, *int32) {
+		var count int32
+		handler := func(resp http.ResponseWriter, req *http.Request) {
+			if req.URL.Path != "/v1/logout" {
+				resp.WriteHeader(http.StatusBadRequest)
+				return
+			}
+			atomic.AddInt32(&count, 1)
+			resp.WriteHeader(http.StatusUnauthorized)
+			_, _ = resp.Write([]byte(`{}`)) // API client requires a JSON response
+		}
+
+		srv := httptest.NewTLSServer(http.HandlerFunc(handler))
+		unauthorizedSrvURL = srv.Listener.Addr().String()
```

Comment:
return this from the function instead of using shared variables

Reply:
done